File: mpi_c/Waermeleitung/dist_index.c
#include <stdio.h>

void dist_index( int n, int *firstind , int nproc)
{
   int  nl, ip;
   
#ifdef DEBUG
   printf("DEBUG: dist_index\t%d\n", nproc);
#endif

   nl = (n + nproc-1) / nproc;
   firstind[0] = 1;
   for (ip = 1; ip <= nproc; ip++)
      firstind[ip] = n+1 < firstind[ip-1]+nl ? n+1 : firstind[ip-1]+nl;
}

File: Ritz/dist_index.py
from mpi4py import MPI
import math
import time
import numpy as np

comm = MPI.COMM_WORLD
nproc= comm.Get_size()
myid = comm.Get_rank()

def dist_index(n):
   nl = int((n+nproc-1) / nproc)
   firstind = np.zeros(nproc+1,dtype=int)
   firstind[0] = 0
   for ip in np.arange(1,nproc+1):
      firstind[ip] = min(firstind[ip-1]+nl,n)
   return firstind
